synfig-studio/src/gui/app.cpp

```diff
@@ -1346,11 +1347,16 @@ App::get_default_accel_map()
 
 	return default_accel_map;
 }
+Glib::RefPtr<App> App::create() {
```

Comment:
I suggest you to rewrite this part like this:
```
Glib::RefPtr<App> App::instance() {
	static Glib::RefPtr<App>app_reference = Glib::RefPtr<App>(new App("", nullptr, nullptr));
	return app_reference;
}
```

in that case you will get standard `singleton` object.

Reply:
Understood ! Changes will be made in the next commit :)

Reply:
@ice0 Gtk::Application class has the following methods:

return type | method signature
--|--
int | run (int argc, char** argv)
int | run (Window& window, int argc, char** argv)
int | run (Window& window)
int | run ()

and


return type | method signature
--|--
static Glib::RefPtr< Application > | create (const Glib::ustring& application_id=Glib::ustring(), Gio::ApplicationFlags flags=Gio::APPLICATION_FLAGS_NONE)
static Glib::RefPtr< Application > | create (int& argc, char**& argv, const Glib::ustring& application_id=Glib::ustring(), Gio::ApplicationFlags flags=Gio::APPLICATION_FLAGS_NONE)

I think we should override these methods.
The  app_basename of our old constructor is taken directly from system calls or arg[0], so we don't have to pass anything to constructor or create() if we pass argc/argv